packages/react/src/components/Rating/Rating.base.tsx

```diff
@@ -139,8 +140,9 @@ export const RatingBase: React.FunctionComponent<IRatingProps> = React.forwardRe
           onFocus={onSelectStar}
           onClick={onSelectStar} // For Safari & Firefox on OSX
           disabled={!!(disabled || readOnly)}
-          role="presentation"
+          role={!readOnly ? 'radio' : 'presentation'}
           type="button"
+          aria-checked={!readOnly ? (starNum === Math.ceil(displayRating) ? true : false) : undefined}
```

Comment:
We don't need the `readOnly` conditional here either anymore, after switching to a conditional `aria-hidden`